src/interfaces/token.ts

```diff
@@ -123,15 +127,63 @@ export class TokenInterface<C extends ChainId> extends ServiceInterface<C> {
   }
 
   /**
-   * Approve vault to spend a token on zapIn
+   * Get allowance for a token either zapIn or direct deposit
    * @param vault
    * @param token
    * @param amount
    * @param account
    * @returns transaction
    */
-  async approve(
-    vault: Vault,
+  async allowance(
+    vault: { address: string; token: string },
```

Comment:
Can we separate this two as different parameters, and not as an object?

Reply:
fixed on 4a954ae5d06e2e712fb3bc2e17b9c73c4a96a565